python/simulation.cpp

```diff
@@ -87,7 +87,14 @@ class simulation_shim {
 
     void record(spike_recording policy) {
         auto spike_recorder = [this](const std::vector<arb::spike>& spikes) {
+            auto old_size = spike_record_.size();
+            // Append the new spikes to the end of the spike record.
             spike_record_.insert(spike_record_.end(), spikes.begin(), spikes.end());
+            // Sort the newly appended spikes.
+            std::sort(spike_record_.begin()+old_size, spike_record_.end(),
+                    [](const auto& lhs, const auto& rhs) {
+                        return std::tie(lhs.time, lhs.source.gid, lhs.source.index)<std::tie(rhs.time, rhs.source.gid, rhs.source.index);
```

Comment:
What about implementing `<=>` for `arb::spike`?

Reply:
I'd prefer not to, but only because the ordering is not natural: we could have just as easily sorted first on GID, for example.